src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/FileWatchers/SolutionConfigFileWatcher.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.IO;
+using NuGet.Configuration;
+
+namespace NuGet.PackageManagement.VisualStudio.Utility.FileWatchers
+{
+    internal sealed class SolutionConfigFileWatcher : IFileWatcher
+    {
+        private bool _disposed;
+        private FileSystemWatcher _solutionWatcher;
+        private FileSystemWatcher? _dotNuGetWatcher;
+
+        public SolutionConfigFileWatcher(string solutionDirectory)
+        {
+            string? current = solutionDirectory.EndsWith("\\", StringComparison.Ordinal)
+                ? Path.GetDirectoryName(solutionDirectory)
+                : solutionDirectory;
+
+            _solutionWatcher = Create(solutionDirectory);
+
+            var dotNuGetPath = Path.Combine(solutionDirectory, NuGetConstants.NuGetSolutionSettingsFolder);
+            if (Directory.Exists(dotNuGetPath))
+            {
+                _dotNuGetWatcher = Create(dotNuGetPath);
+            }
+
+            FileSystemWatcher Create(string path)
+            {
+                var watcher = new FileSystemWatcher(path, Settings.DefaultSettingsFileName);
```

Comment:
What happens here if solution dir doesn't have any nuget.config file at all?

Reply:
The file watcher also raises an event when a solution level NuGet.Config file is created so that settings can be refreshed.

Reply:
Yes, that's right. In fact, the subscription to created events is on the very next line: 
https://github.com/NuGet/NuGet.Client/blob/e386036500c00560b2fe377b99caee61fa9cad89/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/FileWatchers/SolutionConfigFileWatcher.cs#L35